packages/integrations/src/awslambda.ts

```diff
@@ -0,0 +1,255 @@
+import { EventProcessor, Hub, Integration, Scope } from '@sentry/types';
+
+interface AWSLambdaContext {
+  getRemainingTimeInMillis: () => number;
+  callbackWaitsForEmptyEventLoop: boolean;
+  awsRequestId: string;
+  functionName: string;
+  functionVersion: string;
+  invokedFunctionArn: string;
+  logGroupName: string;
+  logStreamName: string;
+}
+
+interface Module {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  exports: any;
+  id: string;
+  filename: string;
+  loaded: boolean;
+  parent: Module | null;
+  children: Module[];
+  path: string;
+  paths: string[];
+}
+/**
+ * NodeJS integration
+ *
+ * Provides a mechanism for NodeJS
+ * that raises an exception for handled, unhandled, timeout and similar times of error
+ * and captures the same in Sentry Dashboard
+ */
+export class AWSLambda implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'AWSLambda';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = AWSLambda.id;
+
+  /**
+   * context.
+   */
+  private _awsContext!: AWSLambdaContext;
+
+  /**
+   * timeout flag.
+   */
+  private _timeoutWarning?: boolean = false;
+
+  /**
+   * @inheritDoc
+   */
+
+  /**
+   * flush time in milliseconds to set time for flush.
+   */
+  private _flushTime?: number;
+
+  public constructor(options: { context?: AWSLambdaContext; timeoutWarning?: boolean; flushTimeout?: number } = {}) {
+    if (options.context) {
+      this._awsContext = options.context;
+    }
+
+    this._timeoutWarning = options.timeoutWarning;
+    this._flushTime = options.flushTimeout;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    const lambdaBootstrap: Module | undefined = require.main;
+
+    if (!this._awsContext && !lambdaBootstrap) {
+      return;
+    }
+    /** configured time to timeout error and calculate execution time */
+    const configuredTimeInMilliseconds =
+      this._awsContext.getRemainingTimeInMillis && this._awsContext.getRemainingTimeInMillis();
+
+    /** rapid runtime instance */
+    // const rapidRuntime = lambdaBootstrap?.children[0].exports;
+    let rapidRuntime;
+    try {
+      rapidRuntime = lambdaBootstrap?.children[0].exports;
+    } catch (err) {
+      rapidRuntime = {};
+    }
+
+    /** handler that is invoked in case of unhandled and handled exception */
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const originalPostInvocationError = rapidRuntime?.prototype?.postInvocationError;
```

Comment:
It'll fail if `rapidRuntime` will be assigned in `catch` statement, because `prototype` will be undefined.
Also please don't use `?` in the production code. TypeScript compiler has those warnings for a reason.

Reply:
Fixed these changes in the latest pull request.